src/fastmcp/server/proxy.py

```diff
@@ -80,18 +93,11 @@ async def list_tools(self) -> list[Tool]:
 
     async def call_tool(self, key: str, arguments: dict[str, Any]) -> ToolResult:
         """Calls a tool, trying local/mounted first, then proxy if not found."""
-        try:
-            # First try local and mounted tools
-            return await super().call_tool(key, arguments)
-        except NotFoundError:
-            # If not found locally, try proxy
-            client = self.client_factory()
-            async with client:
-                result = await client.call_tool(key, arguments)
-                return ToolResult(
-                    content=result.content,
-                    structured_content=result.structured_content,
-                )
+
+        if tool := await self.get_tool(key):
```

Comment:
This is a little too reductive -- the `super().call_tool` method contains all error handling that tool.run() lacks.